### PR TITLE
Add support for isNull and isNotNull conditions

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds a support for `isNull` and `isNotNull` conditions for conditional mutations.

We can specify `isNull` and `isNotNull` conditions as follows:
```java
PutIf putIf =
    ConditionBuilder.putIf(ConditionBuilder.column("c1").isNullInt())
        .and(ConditionBuilder.column("c2").isNullText())
        .build();
```

Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/ConditionalExpression.java [hunk 1]
```diff
@@ -340,5 +340,7 @@ public int hashCode() {
     GTE,
     LT,
     LTE,
+    IS_NULL,
+    IS_NOT_NULL
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/ConditionalExpression.java:344]: Added the new operators, `IS_NULL` and `IS_NOT_NULL`.

File: core/src/main/java/com/scalar/db/io/BigIntColumn.java [hunk 2]
```diff
@@ -79,12 +77,12 @@ public Object getValueAsObject() {
...
   }
 
-  @SuppressWarnings("UnstableApiUsage")
   @Override
   public int compareTo(Column<Long> o) {
     return ComparisonChain.start()
         .compare(getName(), o.getName())
-        .compare(getValue(), o.getValue(), Comparators.emptiesLast(Comparator.naturalOrder()))
+        .compareTrueFirst(hasNullValue(), o.hasNullValue())
+        .compare(getBigIntValue(), o.getBigIntValue())
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/io/BigIntColumn.java:85]: I made small changes for the `compareTo` methods of `XXXColumn` in this PR.

File: core/src/main/java/com/scalar/db/storage/cosmos/ValueBinder.java [hunk 3]
```diff
@@ -1,100 +1,68 @@
...
 
 /**
- * A visitor class to bind {@code Value}s to a {@link StringBuilder}
+ * A visitor class to bind {@code Value}s to a {@link StringBuilder}.
  *
  * @author Yuji Ito
  */
 @NotThreadSafe
-public final class ValueBinder implements ValueVisitor {
+public final class ValueBinder implements ColumnVisitor {
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/storage/cosmos/ValueBinder.java:21]: Are these intended changes for this PR? (looks like coming from a previous PR or it's left unchanged in the previous PR?)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Left one comment about the seemingly unrelated changes in the PR.

File: core/src/main/java/com/scalar/db/storage/cosmos/ValueBinder.java (hunk 3 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/storage/cosmos/ValueBinder.java:21]: Thank you for taking at this PR!

In the previous PR, I didn't remove all the usages of `ValueVisitor`, and I thought I was going to do that in another PR. But the changes were needed for this `isNull` and `isNotNull` conditions support, so I did it in this PR. I will remove the rest of the usages of `ValueVisitor` in another PR. Thanks.

**Pull request merged**